pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -74,9 +81,43 @@ let
       rev = "v${version}";
       hash = "sha256-LVLEn+e7c8013pwiLzJiiIObyrlbBHYaioO/SWbItPQ=";
     };
-    });
+  });
+
+  cutlass = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "cutlass";
+    rev = "v3.0.0";
+    sha256 = "sha256-YPD5Sy6SvByjIcGtgeGH80TEKg2BtqJWSg46RvnJChY=";
+  };
+
+  cuda-common-redist = with cudaPackages; [
+    cuda_cccl # cub/cub.cuh
+    libcublas # cublas_v2.h
+    libcurand # curand.h
+    libcusparse # cusparse.h
+    libcufft # cufft.h
+    cudnn # cudnn.h
+  ];
+
+  cuda-native-redist = symlinkJoin {
+    name = "cuda-native-redist-${cudaVersion}";
+    paths = with cudaPackages; [
+      cuda_cudart # cuda_runtime.h cuda_runtime_api.h
+      cuda_nvcc
+    ] ++ cuda-common-redist;
```

Comment:
Only nvcc goes in nativeBuildInputs (=taken from the build platform's splice), cuda libraries (including cudart) go in buildInputs (=taken from the host platform's packages)

Reply:
By default, onnxruntime will static link CUDA runtime, which results in a big size around 265Mb, and could run without `cuda_cudart`.

I had tried to simply add `-DBUILD_SHARED_LIBS=ON`, and results in errors below, no further in-depth study.

```
onnxruntime> /nix/store/74y3751gsixaz9797ib0hp7c658sp1y5-binutils-2.40/bin/ld: /tmp/nix-build-onnxruntime-1.15.1.drv-11/cctk5oN3.ltrans54.ltrans.o: in function `onnxruntime::TfIdfVectorizer::TfIdfVectorizer(onnxruntime::OpKernelInfo const&)':
onnxruntime> <artificial>:(.text._ZN11onnxruntime15TfIdfVectorizerC2ERKNS_12OpKernelInfoE+0x94c): undefined reference to `onnx::AttributeProto_AttributeType_Name[abi:cxx11](onnx::AttributeProto_AttributeType)'
onnxruntime> /nix/store/74y3751gsixaz9797ib0hp7c658sp1y5-binutils-2.40/bin/ld: <artificial>:(.text._ZN11onnxruntime15TfIdfVectorizerC2ERKNS_12OpKernelInfoE+0x95b): undefined reference to `onnx::AttributeProto_AttributeType_Name[abi:cxx11](onnx::AttributeProto_AttributeType)'
onnxruntime> /nix/store/74y3751gsixaz9797ib0hp7c658sp1y5-binutils-2.40/bin/ld: <artificial>:(.text._ZN11onnxruntime15TfIdfVectorizerC2ERKNS_12OpKernelInfoE+0xb0c): undefined reference to `onnx::AttributeProto_AttributeType_Name[abi:cxx11](onnx::AttributeProto_AttributeType)'
onnxruntime> /nix/store/74y3751gsixaz9797ib0hp7c658sp1y5-binutils-2.40/bin/ld: <artificial>:(.text._ZN11onnxruntime15TfIdfVectorizerC2ERKNS_12OpKernelInfoE+0xb1b): undefined reference to `onnx::AttributeProto_AttributeType_Name[abi:cxx11](onnx::AttributeProto_AttributeType)'
onnxruntime> /nix/store/74y3751gsixaz9797ib0hp7c658sp1y5-binutils-2.40/bin/ld: /tmp/nix-build-onnxruntime-1.15.1.drv-11/cctk5oN3.ltrans99.ltrans.o: in function `onnxruntime::common::Status onnxruntime::OpNodeProtoHelper<onnxruntime::ProtoHelperNodeContext>::GetAttrsAsSpan<long>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, gsl::span<long const, 18446744073709551615ul>&) const':
onnxruntime> <artificial>:(.text._ZNK11onnxruntime17OpNodeProtoHelperINS_22ProtoHelperNodeContextEE14GetAttrsAsSpanIlEENS_6common6StatusERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERN3gsl4spanIKT_Lm18446744073709551615EEE+0xac): undefined reference to `onnx::AttributeProto_AttributeType_Name[abi:cxx11](onnx::AttributeProto_AttributeType)'
onnxruntime> /nix/store/74y3751gsixaz9797ib0hp7c658sp1y5-binutils-2.40/bin/ld: /tmp/nix-build-onnxruntime-1.15.1.drv-11/cctk5oN3.ltrans99.ltrans.o:<artificial>:(.text._ZNK11onnxruntime17OpNodeProtoHelperINS_22ProtoHelperNodeContextEE14GetAttrsAsSpanIlEENS_6common6StatusERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERN3gsl4spanIKT_Lm18446744073709551615EEE+0xc8): more undefined references to `onnx::AttributeProto_AttributeType_Name[abi:cxx11](onnx::AttributeProto_AttributeType)' follow
onnxruntime> collect2: error: ld returned 1 exit status
onnxruntime> make[2]: *** [CMakeFiles/onnxruntime.dir/build.make:160: libonnxruntime.so.1.15.1] Error 1
onnxruntime> make[1]: *** [CMakeFiles/Makefile2:1261: CMakeFiles/onnxruntime.dir/all] Error 2
```

Reply:
Moved `cuda_cudart` to `buildInputs`.